### DISCUSSION TITLE
What should be the value of an exemplar

### DISCUSSION THREAD

**Discussion by stephenafamo** [prometheus/client_golang#991] [Q&A]
I've been reading through the code and documentation but I am not quite sure what the value of an exemplar should be, especially for a counter metric.

For example, the Counter metric implements the [ExemplarAdder](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#ExemplarAdder) interface which looks like this:

```go
type ExemplarAdder interface {
	AddWithExemplar(value float64, exemplar Labels)
}
```

But when using this method, what should be this `float64` value?

**Comment by chancez** [comment]
I was just looking at this myself and discovered the value of the exemplar is the value used in the increment, not the value of the counter. So If your always using `1` to increment an exemplar counter, then the exemplar value is always `1`, while the actual counter continues to go up.
